slick.dataview.js

```diff
@@ -542,36 +556,40 @@
       refresh();
     }
 
-    /**
-     * @param varArgs Either a Slick.Group's "groupingKey" property, or a
-     *     variable argument list of grouping values denoting a unique path to the row.  For
-     *     example, calling collapseGroup('high', '10%') will collapse the '10%' subgroup of
-     *     the 'high' group.
-     */
-    function collapseGroup(varArgs) {
+    function collapseGroup() {
       var args = Array.prototype.slice.call(arguments);
       var arg0 = args[0];
-      if (args.length == 1 && arg0.indexOf(groupingDelimiter) != -1) {
-        expandCollapseGroup(arg0.split(groupingDelimiter).length - 1, arg0, true);
+      var groupingKey;
+      var level;
+
+      if (args.length === 1 && arg0.indexOf(groupingDelimiter) !== -1) {
+        groupingKey = arg0;
+        level = arg0.split(groupingDelimiter).length - 1;
       } else {
-        expandCollapseGroup(args.length - 1, args.join(groupingDelimiter), true);
+        groupingKey = args.join(groupingDelimiter);
+       level = args.length - 1;
       }
+
+      expandCollapseGroup(level, groupingKey, true);
+      onGroupCollapsed.notify({level:level, groupingKey:groupingKey});
     }
 
-    /**
-     * @param varArgs Either a Slick.Group's "groupingKey" property, or a
-     *     variable argument list of grouping values denoting a unique path to the row.  For
-     *     example, calling expandGroup('high', '10%') will expand the '10%' subgroup of
-     *     the 'high' group.
-     */
-    function expandGroup(varArgs) {
```

Comment:
same as what I wrote earlier, plz don't remove the comment and args, they serve a purpose and are valid.

Reply:
Having unused parameters is confusing tbh. Looks like it's just been left or missed, rather than deliberate.

Reply:
not if you call the method and your editor shows you the `varArgs` and the comment with intellisense, then you'll be happy to have it

Reply:
In that case, should you not be using the named parameter within the function?

Reply:
you can't, it's using `arguments` because it can be 1 or more and `arguments` is the standard way to call it. It just sux for commenting purposes though and that is why we have this conversation.

Reply:
OK I'll add it back in. Will be later today though before I can get back to it.